C# 11 required properties do not compile on dotnet 6

[Update(jcouv):] The ask is that required properties not be recommended by the IDE on .NET 6 or lower.

### Describe the bug
When I want to use required properties in a project targeting net6.0 with LangVersion set to latest I face some compile errors.

### To Reproduce

1.
```
<Project Sdk="Microsoft.NET.Sdk">

    <PropertyGroup>
        <TargetFramework>net6.0</TargetFramework>
        <ImplicitUsings>enable</ImplicitUsings>
        <Nullable>enable</Nullable>
        <LangVersion>latest</LangVersion>
    </PropertyGroup>

</Project>

namespace Csharp11OnDotnet6;

public class Person
{
    public required string Name { get; init; }
}
```

2.`dotnet build`

### Exceptions (if any)

```
error CS0656: Missing compiler required member 'System.Runtime.CompilerServices.RequiredMemberAttribute..ctor'
error CS0656: Missing compiler required member 'System.Runtime.CompilerServices.CompilerFeatureRequi
redAttribute..ctor'
```

### Expected Behavior: build successful

--- a/src/Analyzers/CSharp/CodeFixes/MakeMemberRequired/CSharpMakeMemberRequiredCodeFixProvider.cs
+++ b/src/Analyzers/CSharp/CodeFixes/MakeMemberRequired/CSharpMakeMemberRequiredCodeFixProvider.cs
@@ -58,6 +58,12 @@ internal sealed class CSharpMakeMemberRequiredCodeFixProvider : SyntaxEditorBase
 
         var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
 
+        if (semanticModel.Compilation.GetBestTypeByMetadataName("System.Runtime.CompilerServices.RequiredMemberAttribute") is null)
+        {
+            // The attribute necessary to support required members is not present
+            return;
+        }
+
         var fieldOrPropertySymbol = semanticModel.GetDeclaredSymbol(node, cancellationToken);
         if (fieldOrPropertySymbol is IPropertySymbol propertySymbol)
         {

--- a/src/Analyzers/CSharp/Tests/MakeMemberRequired/MakeMemberRequiredTests.cs
+++ b/src/Analyzers/CSharp/Tests/MakeMemberRequired/MakeMemberRequiredTests.cs
@@ -8,6 +8,7 @@ using Microsoft.CodeAnalysis.CSharp.CodeFixes.MakeMemberRequired;
 using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
 using Microsoft.CodeAnalysis.Test.Utilities;
 using Microsoft.CodeAnalysis.Testing;
+using Roslyn.Test.Utilities;
 using Xunit;
 
 namespace Microsoft.CodeAnalysis.CSharp.Analyzers.UnitTests.MakeMemberRequired
@@ -117,6 +118,28 @@ namespace Microsoft.CodeAnalysis.CSharp.Analyzers.UnitTests.MakeMemberRequired
             yield return new[] { "internal", "protected" };
             yield return new[] { "internal", "private" };
             yield return new[] { "internal", "private protected" };
+        }
+
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/68478")]
+        public async Task SimpleSetPropertyMissingRequiredAttribute()
+        {
+            var code =
+                """
+                #nullable enable
+                class MyClass
+                {
+                    public string {|CS8618:MyProperty|} { get; set; }
+                }
+                """;
+
+            await new VerifyCS.Test
+            {
+                TestCode = code,
+                FixedCode = code,
+                LanguageVersion = LanguageVersion.CSharp11,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net60,
+            }.RunAsync();
         }
 
         [Fact]

